src/Avalonia.Base/AvaloniaObject.cs

```diff
@@ -675,25 +608,55 @@ protected virtual void OnPropertyChanged(AvaloniaPropertyChangedEventArgs change
         {
         }
 
+        /// <summary>
+        /// Raises the <see cref="PropertyChanged"/> event for a direct property.
+        /// </summary>
+        /// <param name="property">The property that has changed.</param>
+        /// <param name="oldValue">The old property value.</param>
+        /// <param name="newValue">The new property value.</param>
+        /// <param name="priority">The priority of the binding that produced the value.</param>
+        protected void RaisePropertyChanged<T>(
+            DirectPropertyBase<T> property,
+            Optional<T> oldValue,
+            BindingValue<T> newValue,
+            BindingPriority priority = BindingPriority.LocalValue)
+        {
+            RaisePropertyChanged(property, oldValue, newValue, priority, true);
+        }
```

Comment:
Haven't seen this API before. Not sure how it should work, as this notification can be "detached" from the actual property value. As developer can pass fake old and new value properties, which will raise notification, but actual value wasn't changed.
Was it probably intended to be "protected internal"?

Reply:
Will change this to `private protected`.